### DISCUSSION TITLE
You must let us know whether your app uses any Foreground Service permissions.", "status": "PERMISSION_DENIED in Android

### DISCUSSION THREAD

**Discussion by AnanthaRaoK** [codemagic-ci-cd/codemagic-docs#2545] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Preparing build machine Done

Fetching app sources Done
Installing SDKs Done
Set up code signing identities Done
Installing dependencies Done
Building Android  Done
Publishing Getting Error like below

== Gathering artifacts ==

== Publishing artifacts ==

Publishing artifact app-release.aab
Publishing artifact mapping.txt

Publishing app-release.aab to Google Play
Published app-release.aab to track production
Google Play failed to upload artefacts. You must let us know whether your app uses any Foreground Service permissions.: {
    "error": {
        "code": 403,
        "message": "You must let us know whether your app uses any Foreground Service permissions.",
        "status": "PERMISSION_DENIED"
    }
}

Build failed :|

Publishing failed :|
Google Play failed to upload artefacts. You must let us know whether your app uses any Foreground Service permissions.: {
    "error": {
        "code": 403,
        "message": "You must let us know whether your app uses any Foreground Service permissions.",
        "status": "PERMISSION_DENIED"
    }
}

### Expected results

Build and Published successfuly

### Actual results

Publishing failed :| Google Play failed to upload artefacts. You must let us know whether your app uses any Foreground Service permissions.: { "error": { "code": 403, "message": "You must let us know whether your app uses any Foreground Service permissions.", "status": "PERMISSION_DENIED" } }

**Comment by dtrdic** [comment]
Hi @AnanthaRaoK, there are new foreground permission [requirements](https://support.google.com/googleplay/android-developer/answer/13392821?sjid=10403373841528529919-EU) for submitting applications to the Google Play Store with Android SDK 34 (Android 14). I found users online using [this workaround](https://github.com/expo/expo/issues/26846#issuecomment-1957291582), where they block the permission in app.config.js, and they were able to submit their Android builds. Please check it out

**Comment by lavkushry** [comment]
i also facing same issue i have only these files in Android manifest 
<uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
    <uses-permission android:name="android.permission.RECORD_AUDIO" />
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
and getting Error like

Publishing failed :| Google Play failed to upload artefacts. You must let us know whether your app uses any Foreground Service permissions.: { "error": { "code": 403, "message": "You must let us know whether your app uses any Foreground Service permissions.", "status": "PERMISSION_DENIED" } } 

and There is no any sensitivePermission like thing in privacy 
Declaration
Last edited
Health apps
May 20, 2024
Data safety
Dec 1, 2023
Financial features
Oct 18, 2023
App access
Feb 15, 2023
All or some functionality is restricted
Location permissions
Feb 7, 2023
Advertising ID
Feb 7, 2023
News apps
Feb 7, 2023
Government apps
Feb 7, 2023
Target audience and content
Feb 7, 2023
Content ratings
Feb 7, 2023
Ads
Feb 7, 2023
Privacy policy
 
Please help

**Reply by dtrdic** [reply]
Check the merged manifest, the permission is probably coming from a library you used.

You can see it directly on Android studio https://developer.android.com/build/manage-manifests#inspect_the_merged_manifest_and_find_conflicts (check [here](https://stackoverflow.com/questions/77957857/why-does-my-app-need-android-permission-foreground-service-location) for reference)

Check also [this guide](https://developer.android.com/about/versions/14/changes/fgs-types-required) which provides an example of a foreground service type declaration in the manifest

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.

**Comment by NiladriPadhy** [comment]
This issue is closed as unanswered. So created similar issue for find the solution.
https://github.com/orgs/codemagic-ci-cd/discussions/2719
